src/main/java/org/broadinstitute/dsm/db/dao/kit/BSPKitDao.java

```diff
@@ -0,0 +1,231 @@
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_UPDATE_DUMMY_KIT)) {
+                stmt.setString(1, kitLabel);
+                stmt.setString(2, dsmKitRequestId);
+                int result = stmt.executeUpdate();
+                if (result == 1) {
+                    logger.info("Updated dummy kit, set KitLabel " + kitLabel + " for kit with dsmKitRequestId " + dsmKitRequestId);
+                }
+                else {
+                    throw new RuntimeException("Error updating kit  label for " + dsmKitRequestId + " updated " + result + " rows");
+                }
+            }
+            catch (SQLException e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (results.resultException != null) {
+            throw new RuntimeException("Error updating kit  label for " + dsmKitRequestId, results.resultException);
+        }
+    }
+
+    public static String getRandomParticipantIdForStudy(String ddpInstanceId) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_SELECT_RANDOM_PT)) {
+                stmt.setString(1, ddpInstanceId);
+                ResultSet rs = stmt.executeQuery();
+                if (rs.next()) {
+                    dbVals.resultValue = rs.getString(DBConstants.DDP_PARTICIPANT_ID);
+                }
+            }
+            catch (SQLException e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (results.resultException != null) {
+            throw new RuntimeException("Problem getting a random participant id for instance id " + ddpInstanceId, results.resultException);
+        }
+        if (results.resultValue != null) {
+            return (String) results.resultValue;
+        }
+        return null;
+    }
+
+    public static String getCollaboratorSampleId(int kitTypeId, String participantCollaboratorId, String kitType) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            String collaboratorSampleId = KitRequestShipping.generateBspSampleID(conn, participantCollaboratorId, kitType, kitTypeId);
+            dbVals.resultValue = collaboratorSampleId;
+            return dbVals;
+        });
+        if (results.resultException != null) {
+            throw new RuntimeException("Error getting Collaborator Sample Id for  " + participantCollaboratorId, results.resultException);
+        }
+        else {
+            return (String) results.resultValue;
+        }
+    }
+
+    public Object getBSPKitInfo(@NonNull String kitLabel, @NonNull Response response, NotificationUtil notificationUtil) {
```

Comment:
method wrapped in `Dao` class which work with `Response` as well, too many responsibilities for one class and method.